model/project_parser.go

```diff
@@ -503,7 +502,14 @@ func (bvt *parserBV) hasSpecificActivation() bool {
 // This assumes that the version may not exist yet; otherwise FindAndTranslateProjectForVersion is equivalent.
 func FindAndTranslateProjectForPatch(ctx context.Context, p *patch.Patch) (*Project, *ParserProject, error) {
 	if p.PatchedParserProject == "" {
-		return FindAndTranslateProjectForVersion(p.Version, p.Project)
+		v, err := VersionFindOneId(p.Version)
```

Comment:
I don't know if there's a strong distinction between a patch vs. a version here, but this logic should supposedly handle both patches and non-patches. The current design has it so that a parser project always start out stored in the DB, and it only get moved into S3 if it exceeds the 16 MB limit after generate.tasks. It's assumed that when a patch is initially submitted, the YAML text file is under 16 MB in size.

There is one glitch in that assumption where [triggered patches stuff the entire parser project for a version into the patch document](https://github.com/evergreen-ci/evergreen/blob/debcb42ebed5327e69ebaf1b23040ce2c33af4e7/units/patch_intent.go#L897-L906), so that will have to be worked around. It's ideally not supposed to do that, but it is what it is.

Reply:
Ok, sounds reasonable. Thanks!